File: src/main/java/com/example/progettoenterprise/dto/AttivitaViaggioDTO.java
package com.example.progettoenterprise.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import jakarta.validation.constraints.*;
import java.time.LocalDateTime;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AttivitaViaggioDTO {

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long id;
    @NotBlank
    private String titolo;

    private String descrizione;
    @NotNull
    private LocalDateTime orarioInizio;
    @NotNull
    private LocalDateTime orarioFine;
    @NotBlank
    private String posizione;

    @Min(value=0)
    private Double costo;

}

File: frontendMobile/app/src/main/java/com/example/enterprisemobile/model/dto/AttivitaViaggioDTO.kt
package com.example.enterprisemobile.model

data class AttivitaViaggioDTO(
    val id: Long,
    val titolo: String,
    val descrizione: String?,
    val orarioInizio: String,
    val orarioFine: String,
    val posizione: String,
    val costo: Double
)